github/repos.go

```diff
@@ -791,6 +797,28 @@ func (s *RepositoriesService) RemoveBranchProtection(ctx context.Context, owner,
 	return s.client.Do(ctx, req, nil)
 }
 
+// UpdateRequiredStatusChecks updates the required status checks for a given protected branch.
+//
+// GitHub API docs: https://developer.github.com/v3/repos/branches/#update-required-status-checks-of-protected-branch
+func (s *RepositoriesService) UpdateRequiredStatusChecks(ctx context.Context, owner, repo, branch string, sreq *RequiredStatusChecksRequest) (*RequiredStatusChecks, *Response, error) {
```

Comment:
Even though the fields are optional, this call does nothing if `sreq` is `nil` so I'm thinking that we should either make `sreq` a value and not a pointer or add a check so that `nil` can not be passed.

Reply:
Should this behavior differ from other functions? I modeled the implementation of this after the existing `UpdateBranchProtection` function in the same file:

```
// UpdateBranchProtection updates the protection of a given branch.
//
// GitHub API docs: https://developer.github.com/v3/repos/branches/#update-branch-protection
func (s *RepositoriesService) UpdateBranchProtection(ctx context.Context, owner, repo, branch string, preq *ProtectionRequest) (*Protection, *Response, error) {
	u := fmt.Sprintf("repos/%v/%v/branches/%v/protection", owner, repo, branch)
	req, err := s.client.NewRequest("PUT", u, preq)
	if err != nil {
		return nil, nil, err
	}

	// TODO: remove custom Accept header when this API fully launches
	req.Header.Set("Accept", mediaTypeProtectedBranchesPreview)

	p := new(Protection)
	resp, err := s.client.Do(ctx, req, p)
	if err != nil {
		return nil, resp, err
	}

	return p, resp, nil
}
```

Reply:
Usually in the GitHub Developer API documentation, they mention the custom media headers on an endpoint-by-endpoint basis. So I'm thinking it doesn't apply here.

Reply:
Yes, agree and removed in latest PR update.

Reply:
Sorry for the delay, @nmiyake!
Hmmm... I see you are right about not checking `preq` for `nil`... seems odd to me but shows precedent, so let's run with it. We could always clean all these up later in a subsequent PR. This one has been held up long enough.